### PR TITLE
Feature flag cleanup

### PR DESCRIPTION
**Pull request by pimeys**
- Remove `xml` and `array` flags
- Remove `full` and `single` variants
- `json-1` => `json`
- `uuid-0_8` => `uuid`
- `chrono-0_4` => `chrono`
- Test more feature combinations on CI
- Deny warnings on CI

Closes: https://github.com/prisma/quaint/issues/216

### CODE REVIEW

File: .buildkite/docker.sh [hunk 1]
```diff
@@ -27,11 +27,12 @@ docker run --name test-mssql --network test-net \
     -d mcr.microsoft.com/mssql/server:2019-latest
 
 docker run -w /build --network test-net -v $BUILDKITE_BUILD_CHECKOUT_PATH:/build \
+    -e RUSTFLAGS="-D warnings" \
```
Comment (yoshuawuyts) [inline .buildkite/docker.sh:30]: nice!

**Review by yoshuawuyts — APPROVED** [review]
Implementation-wise this looks good; but probably need someone else to sign off on the direction

File: Cargo.toml [hunk 2]
```diff
@@ -130,3 +119,19 @@ optional = true
...
+optional = true
+
+[dependencies.tokio]
+version = "0.2"
+features = ["rt-threaded", "macros", "sync"]
+optional = true
+
+[dependencies.tokio-util]
+version = "0.3"
+features = ["compat"]
```
Comment (tomhoule) [inline Cargo.toml:136]: Is this still needed?

Comment (pimeys) [reply Cargo.toml:136]: For tinerius

Comment (tomhoule) [reply Cargo.toml:136]: tiberius runs on tokio 0.2?

Comment (pimeys) [reply Cargo.toml:136]: It's `tokio-util`, not `tokio`. Latest `tokio-util` is 0.4!

**Pull request merged**